### PR TITLE
Organise ES7 test and test for more features from tc39/ecma262

### PR DESCRIPTION
**Pull request by thomasfosterau**
As per #490 this pull request adds a number of tests for proposals (at various stages) for ES7 from the tc39/ecma262 repo, and gives each ES7 a category corresponding to their proposal's stage (either `strawman`, `proposal` or `draft`). In addition there is a number of additional tests for subfeatures of the SIMD and Parallel Javascript proposals.

Apologies but I couldn't get the build.js script to run on my system - not a problem with this project, rather Windows 7 insists on opening .js files in an editor rather than let them run using node.

| Feature | Previously | Now |
| --- | --- | --- |
| [Exponentiation operator](https://gist.github.com/rwaldron/ebe0f4d2d267370be882) | :+1: | :+1: |
| [Array comprehensions](http://wiki.ecmascript.org/doku.php?id=harmony:array_comprehensions) | :+1: | :+1: |
| [Generator comprehensons](http://wiki.ecmascript.org/doku.php?id=harmony:array_comprehensions) | :+1: | :+1: |
| [Descructuring comprehensions](https://bugzilla.mozilla.org/show_bug.cgi?id=980828) | :+1: | :+1: |
| [Object rest properties](https://github.com/sebmarkbage/ecmascript-rest-spread) | :+1: | :+1: |
| [Object spread properties](https://github.com/sebmarkbage/ecmascript-rest-spread) | :+1: | :+1: |
| [Async functions](https://github.com/lukehoban/ecmascript-asyncawait) | :+1: | :+1: |
| [Arrow async functions](https://github.com/lukehoban/ecmascript-asyncawait) | :+1: | :+1: |
| [Typed objects](https://github.com/dslomov-chromium/typed-objects-es7) | :+1: | :+1: |
| [Object.observe](http://wiki.ecmascript.org/doku.php?id=harmony:observe) | :+1: | :+1: |
| [Object.getOwnPropertyDescriptors](https://gist.github.com/WebReflection/9353781) | :+1: | :+1: |
| [Array.prototype.includes](https://github.com/tc39/Array.prototype.includes/blob/master/spec.md) | :+1: | :+1: |
| [String.prototype.at](https://github.com/mathiasbynens/String.prototype.at) | :+1: | :+1: |
| [Reflect.Realm](https://gist.github.com/dherman/7568885) | :+1: | :+1: |
| [SIMD](https://github.com/johnmccutchan/ecmascript_simd) | :+1: | :+1: |
| [Parallel JavaScript](http://wiki.ecmascript.org/doku.php?id=strawman:data_parallelism) |  | :+1: |
| [Async generators](https://github.com/jhusain/asyncgenerator) |  | :+1: |
| [Trailing commas in function call expressions](https://github.com/tc39/tc39-notes/raw/master/es6/2014-09/trailing_comma_proposal.pdf) |  | :+1: |
| [ArrayBuffer.transfer](https://gist.github.com/lukewagner/2735af7eea411e18cf20) |  | :+1: |
| [Class and Property Decorators](https://github.com/wycats/javascript-decorators/blob/master/README.md) |  | :+1: |
| [Defensible classes](http://wiki.ecmascript.org/doku.php?id=strawman:defensible_classes) |  |  |
| [Relationships](http://wiki.ecmascript.org/doku.php?id=strawman:relationships) |  |  |
| [String padding](http://wiki.ecmascript.org/doku.php?id=strawman:string_padding) |  | :+1: |
| [Structured Clone](https://github.com/dslomov-chromium/ecmascript-structured-clone) |  |  |
| [Weak References](http://wiki.ecmascript.org/doku.php?id=strawman:weakreferences) |  |  |
| [Set/Map.prototype.toJSON](https://github.com/DavidBruant/Map-Set.prototype.toJSON) |  | :+1: |
| [Class property decorators](https://gist.github.com/jeffmo/054df782c05639da2adb) |  | :+1: |

### CODE REVIEW

File: data-es7.js [hunk 1]
```diff
@@ -479,56 +438,596 @@ exports.tests = [
...
+  name: 'Class and Property Decorators',
+  category: 'proposal',
+  link: 'https://github.com/wycats/javascript-decorators',
+  exec: function(){/*
+    class Person {
+      constructor(){
+        this.children = ['Tom', 'Dick', 'Harry'];
+      }
+      @nonenumerable
+      get kidCount() { return this.children.length; }
```
Comment (zloirock) [inline data-es7.js]: Strange test. Class methods are not enumerable by default.

File: data-es7.js [hunk 2]
```diff
@@ -479,56 +438,596 @@ exports.tests = [
...
+    }
+
+    function nonenumerable(target, name, descriptor) {
+      descriptor.enumerable = false;
+      return descriptor;
+    }
 
+    var guardian = new Person();
+
+    return Object.getOwnPropertyDescriptor(guardian, 'kidCount').enumerable == false;
```
Comment (zloirock) [inline data-es7.js]: The test is broken. `kidCount` isn't own property.

File: data-es7.js (hunk 1 above)
Comment (thomasfosterau) [inline data-es7.js]: @zloirock I'll make a better test - somehow I got this test to work as expected in babel.js last night but obviously the test is broken having looked over the proposal again.

File: data-es7.js [hunk 3]
```diff
@@ -633,5 +1169,114 @@ exports.tests = [
...
+      }
+    }
+  }
 }
 ];
+
+//Shift annex B features to the bottom
+exports.tests = exports.tests.reduce(function(a,e) {
+  var index = ['parallel','syntax','bindings','functions','classes',
+    'built-ins','built-in extensions','misc'].indexOf(e.category);
```
Comment (webbedspace) [inline data-es7.js]: You need to update these lines to sort the categories by stage.

File: data-es7.js [hunk 4]
```diff
@@ -479,56 +438,596 @@ exports.tests = [
...
+      */},
+      res: {
+        firefox39: true,
+        chrome41:  false
+      }
+    }
+  }
+},
+{
+  name: 'Class and Property Decorators',
```
Comment (webbedspace) [inline data-es7.js]: Just as a guideline, the capitalisation typically used by the current tests is all-lowercase, except for built-in classes' names like `Map` or `Object`.

Comment (thomasfosterau) [inline data-es7.js]: Are you referring to the 'Class' or 'SIMD' test?

Comment (webbedspace) [inline data-es7.js]: GitHub in-line diff notes refer to the single line immediately above the note.

Comment (thomasfosterau) [inline data-es7.js]: Oh right yes sorry, I'll rename the class test.

**Pull request merged**